### PR TITLE
New Package r-reproducible

### PR DESCRIPTION
**Pull request by dorton21**

### CODE REVIEW

**Comment by glennpj** [comment]
This depends on #21433.

**Review by tldahlgren — CHANGES_REQUESTED** [review]
At a minimum there are some flake8 issues needing to be resolved:

- var/spack/repos/builtin/packages/r-reproducible/package.py:15: [W291] trailing whitespace
15
- var/spack/repos/builtin/packages/r-reproducible/package.py:22: [W291] trailing whitespace

File: var/spack/repos/builtin/packages/r-reproducible/package.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    making deeply reproducible and reusable content in R. The two
+    workhorse functions are Cache and prepInputs; these allow for:
+    nested caching, robust to environments, and objects with 
+    environments (like functions); and data retrieval and processing
+    in continuous workflow environments. In all cases, efforts are
+    made to make the first and subsequent calls of functions have
+    the same result, but vastly faster at subsequent times by way of
+    checksums and digesting. Several features are still under active
+    development, including cloud storage of cached objects, allowing
+    for sharing between users. Several advanced options are
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/r-reproducible/package.py:22]:
```suggestion
    for sharing between users. Several advanced options are
```

**Comment by tldahlgren** [comment]
Confirmed the given version is available at the URL.

**Review by glennpj — CHANGES_REQUESTED** [review]
This package also has System Requirements.

SystemRequirements: | 'unrar' (Linux/macOS)

**Comment by dorton21** [comment]
I'm also confused by all the failing tests.

**Comment by glennpj** [comment]
> I'm also confused by all the failing tests.

Some of the tests are failing because this has a dependency on `r-require`, which is not in Spack yet, #21433. For example:
```console
>       assert not missing, "These packages have missing dependencies:\n" + (
            "\n".join(lines)
        )
E       AssertionError: These packages have missing dependencies:
E         r-reproducible: [r-require]
E       assert not {'r-reproducible': set(['r-require'])}

lib/spack/spack/test/package_sanity.py:241: AssertionError
!!!!!!!!!!!!!!!!!!!! Interrupted: stopping after 1 failures !!!!!!!!!!!!!!!!!!!!
== 1 failed, 972 passed, 17 skipped, 18 xfailed, 1 xpassed in 588.17 seconds ===
Error: Process completed with exit code 2.
```

**Review by glennpj — APPROVED** [review]
LGTM

**Comment by glennpj** [comment]
Restarting tests now that #21433 has been merged.

**Review by alalazo — APPROVED** [review]

**Review by glennpj — APPROVED** [review]
I had already approved the changes so I am not sure why another review was requested.

**Comment by glennpj** [comment]
@tldahlgren It looks like this just needs your approval.

**Review by tldahlgren — APPROVED** [review]

**Pull request merged**